src/libwolfboot.c

```diff
@@ -350,12 +350,8 @@ static int RAMFUNCTION trailer_write(uint8_t part, uintptr_t addr, uint8_t val)
     /* Calculate write address */
     addr_write = addr_align - ((!nvm_cached_sector) * NVM_CACHE_SIZE);
 
-    /* Ensure that the destination was erased, or force erase */
-    if (*((uint32_t *)(addr_write + NVM_CACHE_SIZE - sizeof(uint32_t)))
-            != FLASH_WORD_ERASED)
-    {
-        hal_flash_erase(addr_write, NVM_CACHE_SIZE);
-    }
+    /* Ensure that the destination was erased */
```

Comment:
Are you sure always doing the erase of the trailing here (vs conditional before) is the right assumption?

Reply:
this was part of code size reduction, worst case scenario from the old code we call erased on an already erased sector